subscription.tf

```diff
@@ -1,12 +1,51 @@
 locals {
   enable_subscription = length(var.subscribed_log_group_matches) + length(var.subscribed_log_group_excludes) > 0
+  observe_log_groups  = [aws_cloudwatch_log_group.group.name, "/aws/lambda/${var.name}"]
 }
 
 moved {
   from = module.observe_cloudwatch_logs_subscription
   to   = module.observe_cloudwatch_logs_subscription[0]
 }
 
+resource "aws_iam_role" "observe_subscription_filter" {
+  name_prefix        = local.name_prefix
+  description        = "Role for subscription filters"
+  assume_role_policy = <<-EOF
+    {
+      "Version": "2012-10-17",
+      "Statement": [
+        {
+          "Action": "sts:AssumeRole",
+          "Principal": {
+            "Service": "logs.amazonaws.com"
+          },
+          "Effect": "Allow"
+        }
+      ]
+    }
+  EOF
+
+  tags = var.tags
+}
+
+resource "aws_iam_role_policy_attachment" "observe_subscription_filter" {
```

Comment:
Avoid role policy attachments unless you explicitly intend for the caller to be able to attach more policies. Prefer authoritative whenever possible, in this case setting `managed_policy_arns` on the role.